packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -818,25 +820,27 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
 
   void _handleDelete({ required bool forward }) {
     assert(_selection != null);
-    String textBefore = selection!.textBefore(_plainText);
-    String textAfter = selection!.textAfter(_plainText);
-    int cursorPosition = selection!.start;
-    // If not deleting a selection, delete the next/previous character.
-    if (selection!.isCollapsed) {
-      if (!forward && textBefore.isNotEmpty) {
-        final int characterBoundary = previousCharacter(textBefore.length, textBefore);
-        textBefore = textBefore.substring(0, characterBoundary);
-        cursorPosition = characterBoundary;
-      }
-      if (forward && textAfter.isNotEmpty) {
-        final int deleteCount = nextCharacter(0, textAfter);
-        textAfter = textAfter.substring(deleteCount);
+    if (!_readOnly) {
```

Comment:
This can be expressed as an early exit, instead of wrapping the entire contents:

```dart
if (_readOnly) {
  return;
}